### PR TITLE
Fix to directly use WAIT_APPROVAL timeout

### PR DESCRIPTION
**Pull request by nakabonne**
**What this PR does / why we need it**:
We just need to use it directly. It could overflow.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/executor/waitapproval/waitapproval.go [hunk 1]
```diff
@@ -54,8 +54,8 @@ func (e *Executor) Execute(sig executor.StopSignal) model.StageStatus {
 		ticker         = time.NewTicker(5 * time.Second)
 	)
 	defer ticker.Stop()
-
-	timer := time.NewTimer(time.Duration(e.StageConfig.WaitApprovalStageOptions.Timeout) * time.Second)
+	timeout := time.Duration(e.StageConfig.WaitApprovalStageOptions.Timeout)
+	timer := time.NewTimer(timeout)
```
Comment (khanhtc1202) [inline pkg/app/piped/executor/waitapproval/waitapproval.go:58]: just to confirm, the bug caused by this deleted `time.Second` annotation, is that right? 🤔 
(Timeout default value currently set as 6h already ref: [config/deployment](https://github.com/pipe-cd/pipe/blob/master/pkg/config/deployment.go#L25))

File: pkg/app/piped/executor/waitapproval/waitapproval.go [hunk 2]
```diff
@@ -54,8 +54,8 @@ func (e *Executor) Execute(sig executor.StopSignal) model.StageStatus {
 		ticker         = time.NewTicker(5 * time.Second)
 	)
 	defer ticker.Stop()
-
-	timer := time.NewTimer(time.Duration(e.StageConfig.WaitApprovalStageOptions.Timeout) * time.Second)
```
Comment (khanhtc1202) [inline pkg/app/piped/executor/waitapproval/waitapproval.go:58]: just to confirm, the bug caused by this deleted `time.Second` annotation, is that right? 🤔 
(Timeout default value currently set as 6h already ref: [config/deployment](https://github.com/pipe-cd/pipe/blob/master/pkg/config/deployment.go#L25))

**Review by nakabonne — COMMENTED** [review]
I think so. It always overflows and looks like being `time.NewTimer(-315919h31m53.884942336s)` at runtime.
https://play.golang.org/p/WhPnXvUp7Ci

File: pkg/app/piped/executor/waitapproval/waitapproval.go (hunk 2 above)
Comment (nakabonne) [reply pkg/app/piped/executor/waitapproval/waitapproval.go:58]: I think so. It always overflows and looks like being `time.NewTimer(-315919h31m53.884942336s)` at runtime.
https://play.golang.org/p/WhPnXvUp7Ci

Comment (khanhtc1202) [reply pkg/app/piped/executor/waitapproval/waitapproval.go:58]: I see, I met this issue before 😂 ( time.Minute * time.Minute overflow in my case )

Comment (khanhtc1202) [reply pkg/app/piped/executor/waitapproval/waitapproval.go:58]: And one more minor question, it looks like the default value is of type Duration already, is that `time.Duration(...)` is necessary for the new code? 👀

Comment (nakabonne) [reply pkg/app/piped/executor/waitapproval/waitapproval.go:58]: Look closely. The type of `Duration` used by Config is our own Type for making it easier to parse. So we have to parse it to time.Duration when using 😄

Comment (khanhtc1202) [reply pkg/app/piped/executor/waitapproval/waitapproval.go:58]: Yeh, but it a defined type based on `time.Duration` (ref: [config/duration.go](https://github.com/pipe-cd/pipe/blob/master/pkg/config/duration.go#L23) ). Still looks weird to me, but never mind if it works that way 😅

Comment (khanhtc1202) [reply pkg/app/piped/executor/waitapproval/waitapproval.go:58]: I mean
```go
e.StageConfig.WaitApprovalStageOptions.Timeout.Duration()
```
should be fine 😂

Comment (nakabonne) [reply pkg/app/piped/executor/waitapproval/waitapproval.go:58]: Ah, you're right, let me fix.

**Comment by nghialv** [comment]
Nice. Thank you.
/approve

**Pull request merged**